### PR TITLE
dev/core#2422 Add created_id, modified_id, expires_date to saved search table

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Per https://lab.civicrm.org/dev/core/-/issues/2422 sites using search kit will rapidly start to accumulate a large number of saved searches and will want better filtering (& cleanup). This adds created_id and modified_id and starts populating them as well as expires_date (which won't start populating until we expose it in the search kit UI).

Starting to populate this data in 5.36 (before we've had a chance to expose it) means that when we do people will see data not just empty fields.

Before
----------------------------------------
id of saved_search creator / modifier not stored. No field for expires_date

After
----------------------------------------
tada

Technical Details
----------------------------------------
@colemanw 

Comments
----------------------------------------

### CODE REVIEW

**Comment by mattwire** [comment]
@eileenmcnaughton It seems to me if we are adding `created_id` and `modified_id` that it would also make a lot of sense to add `created_date` and `modified_date` at the same time? Extremely useful for troubleshooting and pretty useful if exposed to the UI as well.

**Comment by eileenmcnaughton** [comment]
@mattwire I did consider those - I'm happy to add them in if you want me to (I wavered because they aren't in the civicrm_group table yet I think)

**Comment by mattwire** [comment]
I think it would be good. I've generally found wherever we're recording a create/modify contact_id the date is useful too.

**Comment by eileenmcnaughton** [comment]
@mattwire ok - I've added them in

**Comment by totten** [comment]
I haven't `r-run` - but just want to say it looks like this is generally good practice - and we should merge before branching.

**Comment by eileenmcnaughton** [comment]
after spending more time playing with saved search I think we also want a description field to elaborate on what the searches are (also in the group table).

I realise my biggest challenge now is getting this through review so not sure whether to attempt to add that or wait for this to hopefully get merged

**Comment by colemanw** [comment]
I'm happy to merge this after you add a description field @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
@colemanw should I go whole hog & add frontend_description per Group.xml

**Comment by eileenmcnaughton** [comment]
Well I've added in description now

File: CRM/Contact/DAO/SavedSearch.php [hunk 1]
```diff
@@ -86,6 +86,46 @@ class CRM_Contact_DAO_SavedSearch extends CRM_Core_DAO {
...
+  /**
+   * When the search was created.
+   *
+   * @var timestamp
+   */
+  public $created_date;
+
+  /**
+   * When the search was lase modified.
+   *
```
Comment (aydun) [inline CRM/Contact/DAO/SavedSearch.php:119]: Typo: "lase"

File: CRM/Contact/DAO/SavedSearch.php [hunk 2]
```diff
@@ -250,6 +292,95 @@ public static function &fields() {
...
+          'entity' => 'SavedSearch',
+          'bao' => 'CRM_Contact_BAO_SavedSearch',
+          'localizable' => 0,
+          'add' => '5.36',
+        ],
+        'modified_date' => [
+          'name' => 'modified_date',
+          'type' => CRM_Utils_Type::T_TIMESTAMP,
+          'title' => ts('Modified Date'),
+          'description' => ts('When the search was lase modified.'),
```
Comment (aydun) [inline CRM/Contact/DAO/SavedSearch.php:358]: Typo: "lase"

File: xml/schema/Contact/SavedSearch.xml [hunk 3]
```diff
@@ -136,4 +136,73 @@
...
+    <type>timestamp</type>
+    <comment>When the search was created.</comment>
+    <required>true</required>
+    <default>CURRENT_TIMESTAMP</default>
+    <add>5.36</add>
+  </field>
+  <field>
+    <name>modified_date</name>
+    <type>timestamp</type>
+    <comment>When the search was lase modified.</comment>
```
Comment (aydun) [inline xml/schema/Contact/SavedSearch.xml:192]: Typo: "lase"
At least it's consistent!

File: CRM/Contact/DAO/SavedSearch.php (hunk 2 above)
Comment (totten) [reply CRM/Contact/DAO/SavedSearch.php:358]: This one is actually supposed to be modified with lasers.

**Comment by colemanw** [comment]
> @colemanw should I go whole hog & add frontend_description per Group.xml

I wouldn't, no. SavedSearches are a backend thing. SearchDisplays are more front-endey.

File: CRM/Contact/DAO/SavedSearch.php (hunk 2 above)
Comment (colemanw) [reply CRM/Contact/DAO/SavedSearch.php:358]: I dunno if we should add lasers as a hard-dependency. We want to support older systems that don't have built-in CD burners. IMO Civi should work on any box fast enough to run Win95.

Comment (eileenmcnaughton) [reply CRM/Contact/DAO/SavedSearch.php]: OK - you had your fun - all lasers updated now

**Comment by eileenmcnaughton** [comment]
@colemanw all passing now without the lasers

**Pull request merged**